src/lib/stores/busy.store.ts

```diff
@@ -0,0 +1,48 @@
+import { derived, writable, type Readable } from "svelte/store";
+import type { BusyState } from "$lib/types/busy";
+
+/**
+ * Store that reflects the app busy state.
+ * Is used to show the busy-screen that locks the UI.
+ */
+const initBusyStore = () => {
+  const { subscribe, update } = writable<Array<BusyState<any>>>([]);
+
+  return {
+    subscribe,
+    /**
+     * Show the busy-screen if not visible
+     */
+    startBusy({ initiator: newInitiator, text }: BusyState<any>) {
+      update((state) => [
+        ...state.filter(({ initiator }) => newInitiator !== initiator),
+        { initiator: newInitiator, text },
+      ]);
+    },
+
+    /**
+     * Hide the busy-screen if no other initiators are done
+     */
+    stopBusy(initiatorToRemove: BusyState<any>["initiator"]) {
+      update((state) =>
+        state.filter(({ initiator }) => initiator !== initiatorToRemove)
+      );
+    },
+  };
+};
+
+export const busyStore = initBusyStore();
```

Comment:
ah I see now here we will have an issue with the generic...

you know what, maybe we can just replace it with a `string`

```
export interface BusyState {
  initiator: string;
  text?: string;
}
```

what do you think could that work? in NNS-dapp with use a type that contains `string` so maybe that will do?

Reply:
I can make that change, seems most sensible